### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by YevhenOstrovskiy**

### CODE REVIEW

File: src/App.jsx [hunk 1]
```diff
@@ -14,45 +16,97 @@ export const goodsFromServer = [
...
+    }
+
+    setSortGoods('');
+    setGoods(goodsFromServer);
+  };
+
+  const sortByAlphabetically = () => {
+    setSortGoods('Sort alphabetically');
+    if (isReversed) {
+      setGoods([...goods].sort((a, b) => b.localeCompare(a)));
```
Comment (EdenParadise) [inline src/App.jsx:35]: create const for 
`[...goods]` 
and use it afterwards in your code

File: src/App.jsx [hunk 2]
```diff
@@ -14,45 +16,97 @@ export const goodsFromServer = [
...
+
+      return;
+    }
+
+    setGoods([...goods].sort((a, b) => a.localeCompare(b)));
```
Comment (EdenParadise) [inline src/App.jsx:40]: Same here
create const
`const preparedGoods = [...goods];`

File: src/App.jsx [hunk 3]
```diff
@@ -14,45 +16,97 @@ export const goodsFromServer = [
...
+  };
+
+  const sortByLenght = () => {
+    setSortGoods('Sort by length');
+    if (isReversed) {
+      setGoods([...goods].sort((a, b) => b.length - a.length));
```
Comment (EdenParadise) [inline src/App.jsx:46]: Same here
create const
`const preparedGoods = [...goods];`

**Review by EdenParadise — CHANGES_REQUESTED** [review]
Good job, but need to implement some changes. Keep it up!

File: src/App.jsx [hunk 4]
```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
...
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    goodsCopy.reverse();
+  }
+
+  return goodsCopy;
+}
```
Comment (DanilWeda) [inline src/App.jsx:43]: you can move it outside component file

File: src/App.jsx [hunk 5]
```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
...
+            { 'is-light': !isReversed },
+          )}
+          onClick={() => {
+            setReversed(!isReversed);
+          }}
+        >
+          Reverse
+        </button>
+
+        {(sortGoods.length > 0 || isReversed) && (
```
Comment (DanilWeda) [inline src/App.jsx:101]:
```suggestion
        {(sortGoods.length  || isReversed) && (
```
as alternative*

File: src/App.jsx [hunk 6]
```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
...
+
+        <button
+          type="button"
+          className={classNames(
+            'button is-success',
+            { 'is-light': sortGoods !== SORT_BY_LENGTH },
+          )}
+          onClick={() => {
+            setSortGoods(SORT_BY_LENGTH);
+          }}
```
Comment (DanilWeda) [inline src/App.jsx:83]: you can write a handler for it ( not critical )

File: src/App.jsx [hunk 7]
```diff
@@ -14,45 +16,104 @@ export const goodsFromServer = [
...
+      <div className="buttons">
+        <button
+          type="button"
+          className={classNames(
+            'button is-info',
+            { 'is-light': sortGoods !== SORT_ALPHABETICALLY },
+          )}
+          onClick={() => {
+            setSortGoods(SORT_ALPHABETICALLY);
+          }}
```
Comment (DanilWeda) [inline src/App.jsx:70]: you can write a handler for it ( not critical )

**Review by DanilWeda — APPROVED** [review]
Good job! 
Approved.
I left comments about code splitting, please fix it! 
Thanks.

**Pull request closed without merging**